pulsar-broker/src/main/java/com/yahoo/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -379,6 +381,27 @@ private LookupResult findBrokerServiceUrl(ServiceUnitId suName, boolean authorit
         }
     }
 
+    private LookupResult createLookupResult(String candidateBroker) throws Exception {
+
+        checkArgument(StringUtils.isNotBlank(candidateBroker), "Lookup broker can't be null " + candidateBroker);
+        URI uri = new URI(candidateBroker);
+        String path = String.format("%s/%s:%s", SimpleLoadManagerImpl.LOADBALANCE_BROKERS_ROOT, uri.getHost(),
+                uri.getPort());
+        LoadReport report = pulsar.getLocalZkCache().getData(path, new Deserializer<LoadReport>() {
+            @Override
+            public LoadReport deserialize(String key, byte[] content) throws Exception {
+                return jsonMapper().readValue(content, LoadReport.class);
+            }
+        });
+
+        LookupResult result = new LookupResult(new URI(candidateBroker), report.getPulsarServiceUrl(), report.getPulsarServieUrlTls());
```

Comment:
Actually in this case, returned ```LookupResult``` contains ```redirect``` urls.
- [Http lookup request](https://github.com/rdhabalia/pulsar/blob/fc6fa39dbbcc29fa64efcab25a48ade3d5bc688a/pulsar-broker/src/main/java/com/yahoo/pulsar/broker/lookup/DestinationLookup.java#L71) just requires broker-url where call should be redirected. So,  ```LookupResult``` has one ```http/https url```
-  [Binary-Protocol request] (https://github.com/rdhabalia/pulsar/blob/fc6fa39dbbcc29fa64efcab25a48ade3d5bc688a/pulsar-broker/src/main/java/com/yahoo/pulsar/broker/lookup/DestinationLookup.java#L154) sends both broker service-url/tls and client decides which one needs to be selected based on client configuration.